statsmodels/base/transform.py

```diff
@@ -0,0 +1,217 @@
+            The method by which to estimate lambda. Defaults to 'guerrero', but
+            the profile likelihood ('loglik') is also available.
+        **kwargs : dict
+            Options dict for the specified method.
+            * For 'guerrero': R (int), the seasonality/grouping parameter. Scale
+              ({'mad', 'sd'}), the dispersion measure.
+            * For 'loglik': this is not used.
+
+        Returns
+        -------
+        lmbda : float
+            The lambda parameter.
+        """
+        method = method.lower()
+
+        if len(bounds) != 2:
+            raise ValueError("Bounds of length {0} not understood."
+                             .format(len(bounds)))
+        elif bounds[0] >= bounds[1]:
+            raise ValueError("Lower bound exceeds upper bound.")
+
+        if method == 'guerrero':
+            lmbda = self._guerrero_cv(x, bounds=bounds, **kwargs)
+        elif method == 'loglik':
+            lmbda = self._loglik_boxcox(x, bounds=bounds)
+        else:
+            raise ValueError("Method '{0}' not understood.".format(method))
+
+        return lmbda
+
+    def _guerrero_cv(self, x, bounds, R=4, scale='sd'):
+        """
+        Computes lambda using guerrero's coefficient of variation. If no
+        seasonality is present in the data, R is set to 4 (as per Guerrero
+        and Perera, (2004)).
+
+        NOTE: Seasonality-specific auxiliaries *should* provide their own
+        seasonality parameter.
+
+        Parameters
+        ----------
+        x : array_like
+        bounds : tuple
+            Numeric 2-tuple, that indicate the solution space for the lambda
+            parameter.
+        R : int
+            Seasonality/grouping parameter. Default 4, as per Guerrero and
+            Perera (2004). NOTE: this indicates the length of the individual
+            groups, not the total number of groups!
+        scale : {'sd', 'mad'}
+            The dispersion measure to be used. 'sd' indicates the sample
+            standard deviation, but the more robust 'mad' is also available.
+        """
+        nobs = len(x)
+        groups = int(nobs / R)
+
+        # remove the first n < R observations from consideration.
+        grouped_data = np.reshape(x[nobs - (groups * R): nobs], (groups, R))
+        mean = np.mean(grouped_data, 1)
+
+        scale = scale.lower()
+        if scale == 'sd':
+            dispersion = np.std(grouped_data, 1, ddof=1)
+        elif scale == 'mad':
+            dispersion = mad(grouped_data, axis=1)
+        else:
+            raise ValueError("Scale '{0}' not understood.".format(scale))
+
+        def optim(lmbda, *args, **kwargs):
+            rat = np.divide(dispersion, np.power(mean, 1 - lmbda))  # eq 6, p 40
+            return np.std(rat, ddof=1) / np.mean(rat)
+
+        res = minimize_scalar(optim,
+                              bounds=bounds,
+                              method='bounded',
+                              options={'maxiter': 50})
```

Comment:
I'll make the options object available through the function arguments (as it really should be controllable).